### ISSUE TITLE
Let's get rid of dimension scales!

### ISSUE THREAD

**Issue by edhartnett** [Unidata/netcdf-c#865]
Can we stop using dimension scales? I think they have no fans on the netCDF development team. To put it bluntly: they are a very poor way to encode netCDF dimension information.

Of course, we will always have to read them. But do we have to continue creating files that use them?

Suppose we come up with some special att(s) which encodes the dimension information. When those special att(s) are present in a file, we don't look for dimension scales that are not coordinate variables.

When we create new files, we use our new special atts and stop using dimension scales. When we have a particularly troublesome problem, like some complex rename issue, we can just internally convert the file to the new method, if we need to.

(No this is not just an attempt to avoid figuring out the remaining rename problems!) ;-)

I would not consider tackling this until all other proposed work has been completed. It's just a long-term thought that I wanted to get out there for discussion. Maybe it should not even be attempted.

**Comment by DennisHeimbigner** [comment]
I was thinking the same thing. The big problem is backward compatibility.
It means that we can not get rid of the existing dimension scale code for the
foreseeable future. We also need a clear specification of the new implementation.

**Comment by edhartnett** [comment]
Yes, just putting it on the radar. Still far from proposing an implementation. ;-)

**Comment by DennisHeimbigner** [comment]
In looking at Caron's analysis:
https://www.unidata.ucar.edu/blogs/developer/entry/dimensions_scales
We cannot get rid of dimension scales. The real issue IMO is that we need to
not conflate the dimscale dataset with the coordinate variable dataset.
This apparently requires (mostly) kind kind of special naming convention
that allows us to have two datasets: one for the dimension and one for
the coord variable.

**Comment by DennisHeimbigner** [comment]
So really, this issue should be entitled:  "Let's quit conflating
dimensions and coordinate variable".
That is a lot more doable.

**Comment by edhartnett** [comment]
OK. ;-)

This is not something I am actively thinking about yet though, so don't expect anything soon. ;-)
